kmath-geometry/src/commonMain/kotlin/scientifik/kmath/geometry/EuclideanSpace.kt

```diff
@@ -0,0 +1,57 @@
+package scientifik.kmath.geometry
+
+import scientifik.kmath.dimensions.*
+import scientifik.kmath.operations.ExtendedField
+import scientifik.kmath.operations.Field
+import scientifik.kmath.operations.invoke
+import scientifik.kmath.structures.asSequence
+
+internal data class VectorImpl<T, D : Dimension>(private val array: List<T>) : Vector<T, D> {
+    override val size: Int get() = array.size
+
+    override fun get(index: Int): T = when {
+        index < 0 || index >= size -> error("Accessing outside of point bounds")
+        else -> array[index]
+    }
+
+    override fun iterator(): Iterator<T> = array.iterator()
+}
+
+fun <T> vectorOf(a1: T): Vector<T, D1> = VectorImpl(listOf(a1))
+fun <T> vectorOf(a1: T, a2: T): Vector<T, D2> = VectorImpl(listOf(a1, a2))
+fun <T> vectorOf(a1: T, a2: T, a3: T): Vector<T, D3> = VectorImpl(listOf(a1, a2, a3))
+fun <T> vectorOf(a1: T, a2: T, a3: T, a4: T): Vector<T, D4> = VectorImpl(listOf(a1, a2, a3, a4))
+fun <T> vectorOf(a1: T, a2: T, a3: T, a4: T, a5: T): Vector<T, D5> = VectorImpl(listOf(a1, a2, a3, a4, a5))
```

Comment:
Do we actually have applications for 5d vectors?

Reply:
No, was actually inspired by kotlin grad library, where they created helpers up to 100 dimension,

Reply:
Kotlingrad is intended for many-dimensional operations. In this case, we probably need to cover most used dimensions and leave the rest generic.

Reply:
It is probably better to generate these declarations at compile time using a template/macro. We wrote some crude string-based [code generators](https://github.com/breandan/kotlingrad/tree/e30d8331e6209cb483b021a9759926aaa4d13add/core/src/main/kotlin/edu/umontreal/kotlingrad/utils/codegen), which you might be able to adapt in case you do end up using this strategy. There might be a better way to provide similar guarantees without using macros. If you find one, please let me know!